File: 6计算机控制/1计算机控制性能分析/jury.m
function jury(input)
%myFun - Description
%
% Syntax: jury(input)
%
% Long description
disp('利用朱利判据判断离散系统的稳定性')
disp('请输入特征方程的系数矩阵A')
A=input;
n=size(A,2);%截取A的维数
for i=n:-1:1
    disp(A);%显示奇数行
    if i==1 
        if A(:)>0
            disp('系统稳定');
        else
            disp('系统不稳定');
        end
        break;
    end
    B=flip(A);
    disp(B);%显示偶数行
    if A(1,1)<=0
        disp('首元素非正！系统不稳定');
        break;
    end
    k=B(1,1)/A(1,1);
    A(1,:)=A(1,:)-k*B(1,:);
    A(:,i)=[];%A矩阵减维数
end
% ————————————————
% 版权声明：本文为CSDN博主「非线性光学元件」的原创文章，遵循CC 4.0 BY-SA版权协议，转载请附上原文出处链接及本声明。
% 原文链接：https://blog.csdn.net/weixin_44044411/article/details/86475605
end

File: 6计算机控制/1计算机控制性能分析/jury.py
# 绘制朱利判据表
# 参考https://blog.csdn.net/weixin_44044411/article/details/86475605

print('利用朱利判据判断离散系统的稳定性')
print('请输入特征方程的系数矩阵A')
A=[1,2,1.31,0.28]#input('A=')
n=len(A)#截取A的长度
for i in list(reversed(range(1,n))):
    print(A);#显示奇数行
    if i==1:
        if A[0]>0:
            print('系统稳定')
        else:
            print('系统不稳定')
        break
    B=list(reversed(A))
    print(B)#显示偶数行
    if A[0]<=0:
        print('首元素非正！系统不稳定')
        break
    k=B[0]/A[0]
    for index in range(0,len(A)):
        A[index]=A[index]-k*B[index]
    A.pop()
